tools/google-cloud-support-slackbot/google_cloud_support_slackbot.py

```diff
@@ -0,0 +1,811 @@
+#!/usr/bin/env python3
+
+# Copyright 2021 Google LLC
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import slack
+import os
+import json
+import requests
+import time
+import re
+import logging
+from pathlib import Path
+from dotenv import load_dotenv
+from flask import Flask, request, Response
+from multiprocessing import Process
+from googleapiclient.discovery import build_from_document
+from datetime import datetime
+
+# To run this on the cheapest possible VM, we will only log Warnings and Errors
+logging.basicConfig(filename='error.log')
+flask_logger = logging.getLogger('werkzeug')
+flask_logger.setLevel(logging.WARNING)
+
+logging.warning('Started at: {}'.format(datetime.now()))
+
+# Load our environment variables from the .env file
+env_path = Path('.') / '.env'
+load_dotenv(dotenv_path=env_path)
+
+app = Flask(__name__)
+
+client = slack.WebClient(token=os.environ['SLACK_TOKEN'])
+ORG_ID = os.environ['ORG_ID']
+API_KEY = os.environ['API_KEY']
+
+discovery_doc_file = 'cloudsupport.v2alpha.json'
+
+# Get our discovery doc
+r = requests.get('https://cloudsupport.googleapis.com/$discovery/rest?key={}&labels=V2_TRUSTED_TESTER&version=v2alpha'.format(API_KEY))
+
+# Add the discovery doc, or replace the previous version with the latest one
```

Comment:
I _think_ you should be able to eliminate the need to write/read a file on disk for this discovery doc. As far as I can see you're just using this discovery_doc_file to store the results from the GCP API call, and then reading them back into a variable for further use.

Perhaps something like:
```
r = requests.get('https://cloudsupport.googleapis.com/$discovery/rest?key={}&labels=V2_TRUSTED_TESTER&version=v2alpha'.format(API_KEY))

discovery_doc = build_from_document(json.dump(r.json))
```